Diagnostics/start-windows-build.ps1

```diff
@@ -0,0 +1,405 @@
+}
+
+function Start-TempDirCleanup {
+    Get-ChildItem $env:TEMP | Where-Object {
+        $_.Name -notmatch "^jna\-[0-9]*$|^hsperfdata.*_diagnostics$"
+    } | ForEach-Object {
+        $fullPath = $_.FullName
+        if($_.FullName -is [System.IO.DirectoryInfo]) {
+            cmd.exe /C "rmdir /s /q ${fullPath} > nul 2>&1"
+        } else {
+            cmd.exe /C "del /Q /S /F ${fullPath} > nul 2>&1"
+        }
+    }
+}
+
+function Run-DiagnosticsUnittests {
+    Write-Output "Run Diagnostics unittests"
+    Push-Location
+    Set-Location $DIAGNOSTICS_GIT_REPO_DIR
+    $currLoc = Get-Location
+    write-output "Current location: $($currLoc.Path)" 
+    $logsUrl = Get-BuildLogsUrl
+    try {
+        $parameters = @(".\scripts\make.ps1", "test")
+        $processPath = "powershell.exe"
+        Start-DiagnosticsCIProcess  -ProcessPath  $processPath `
+                                    -StdoutFileName "diagnostics-unitests-stdout.log" `
+                                    -StderrFileName "diagnostics-unitests-stderr.log" `
+                                    -ArgumentList $parameters `
+                                    -BuildErrorMessage "Diagnostics failed to build."
+    } finally {
+        Pop-Location
+    }
+
+    if($global:PARAMETERS["BUILD_STATUS"] -eq 'FAIL') {
+        $errMsg = "Some of the unit tests failed. Please check the relevant logs."
+        $global:PARAMETERS["FAILED_COMMAND"] = 'Run-DiagnosticsUnittests'
+        Throw $errMsg
+    }  
+    Write-Output "All Diagnostics unittests passed successfully"
+}
+
+function New-ParametersFile {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$FilePath
+    )
+    if(Test-Path $FilePath) {
+        Remove-Item -Force $FilePath
+    }
+    New-Item -ItemType File -Path $FilePath | Out-Null
+}
+
+function Write-ParametersFile {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$FilePath
+    )
+    if($global:PARAMETERS["LOGS_URLS"]) {
+        $global:PARAMETERS["LOGS_URLS"] = $global:PARAMETERS["LOGS_URLS"] -join '|'
+    }
+    $json = ConvertTo-Json -InputObject $global:PARAMETERS
+    Set-Content -Path $FilePath -Value $json
+}
+
+try {
+    Start-TempDirCleanup
+    New-ParametersFile -FilePath $ParametersFile
+    Install-Prerequisites
+    Clone-DiagnosticsSource
+    Build-DiagnosticsBinaries
+    Run-DiagnosticsUnittests
+    Generate-DiagnosticsZipFile
+
+    $global:PARAMETERS["BUILD_STATUS"] = "PASS"
+    $global:PARAMETERS["MESSAGE"] = Get-SuccessBuildMessage
+} catch {
+    Write-Output $_.ToString()
+    Write-Output $_.ScriptStackTrace
+    $global:PARAMETERS["BUILD_STATUS"] = "FAIL"
+    $global:PARAMETERS["MESSAGE"] = $_.ToString()
+    exit 1
+} finally {
+    # The following upload does not work for me. I am suspecting it requires some privilege.
+    # For testing other part of the code, I replace it with simple file copying below.
+    #Start-LogServerFilesUpload
```

Comment:
Uncomment `Start-LogServerFilesUpload` method and remove your comment. Don't worry too much about the upload. You won't be able to do it from outside the CI Azure resource group anyway.

I'll make sure this works as expected when adding the script in the CI. So far, I can tell that you followed the correct pattern from Mesos `start-windows-build.ps1` script and this part is aprox 80% done.